testng-core-api/src/main/java/org/testng/IObjectBag.java

```diff
@@ -0,0 +1,27 @@
+package org.testng;
+
+import java.io.Closeable;
+import java.util.function.Supplier;
+
+/**
+ * Represents a bean bag which is scoped to a specific {@link ISuite}. After a {@link ISuite} wraps
+ * up its execution, the contents of this bean bag will be cleaned-up.
+ */
+public interface IObjectBag {
+
+  /**
+   * @param type - The type of the object to be created
+   * @param supplier - A {@link Supplier} that should be used to produce a new instance
+   * @param <T> - The type of the object
+   * @return - Either the newly produced instance or the existing instance.
+   */
+  <T> T createIfRequired(Class<T> type, Supplier<T> supplier);
```

Comment:
`Class<T> type` is useless. The type is already provided by the supplier param.

Reply:
This interface is removed.